features/url-formattering.feature

```diff
@@ -0,0 +1,127 @@
+    en uri formats voor Resource Identifiers (zie [7.3.5. Resource Identifiers](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3.5)
+    en [7.3.6. uri-template](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3.6)
+    in de JSON Schema Validation specificatie) om dit mogelijk te maken.
+
+    Door middel van een custom HTTP header 'Accept-Uri-Format' kan een Haal Centraal API consumer in een request
+    aangeven in welke format hij de urls in de response wil ontvangen.  
+    De volgende uri formats worden ondersteund:
+    - uri. Alle links in de response worden geformatteerd als absolute urls.
+    - uri-reference (default). 'Same API' links worden geformatteerd als relatieve urls en 'externe API' links als absolute urls.
+    - uri-template. De base url van links worden gerepresenteerd door een URI Template [Expression](https://datatracker.ietf.org/doc/html/rfc6570#section-2.2).
+      De Haal Centraal APIs kennen de volgende base url expressions:
+      - {brpserverurl}
+      - {brphistorieserverurl}
+      - {brpbewoningserverurl}
+      - {bagserverurl}
+      - {brkserverurl}
+      - {wozserverurl} 
+
+    De Haal Centraal APIs gebruiken de custom 'Content-Uri-Format' HTTP header om aan te geven welke uri format
+    is gebruikt om de urls in de response te formatteren.    
+
+    De [URI Template](https://datatracker.ietf.org/doc/html/rfc6570) specificatie wordt door de Haal Centraal APIs
+    ook gebruikt om gelijksoortige links naar resources met URI templates te representeren.
+    Een URI template bevat één of meerdere expressions waarmee een verzameling van URIs kan worden beschreven.
+    Een expression is de tekst tussen een open accolade '{' en een sluit accolade '}' inclusief de accolade tekens.
+    Er zijn meerdere expression types met bijbehorende expansie methode. Zie [Expressions](https://datatracker.ietf.org/doc/html/rfc6570#section-2.2).
+    De Haal Centraal APIs gebruiken 'no operator expressions' zodat consumers 'Simple String Expansion'
+    kunnen toepassen. Dit houdt in dat een {expression} zonder toepassing van logica kan worden vervangen met een waarde.
+    Binnen de Haal Centraal APIs worden property namen gebruikt voor expressions. Een property naam is:
+    - de naam van een property van een resource, v.b. burgerservicenummer of
+    - de naam van een property van een gegevensgroep van een resource v.b. kinderen.burgerservicenummer
+
+Achtergrond:
+    * de Haal Centraal APIs met hun base url en base url template placeholder
+    | Naam Haal Centraal API | base url                                                         | base url template placeholder |
+    | BRP Bevragen           | https://www.haalcentraal.nl/haalcentraal/api/brp                 | {brpserverurl}                |
+    | BAG Bevragen           | https://api.bag.acceptatie.kadaster.nl/esd/huidigebevragingen/v1 | {bagserverurl}                |
```

Comment:
het is denk ik niet handig in een feature naar de BAG acceptatieomgeving te verwijzen. Gebruikers gaan niet naar de acceptatieomgeving, die gebruiken alleen productie: https://api.bag.kadaster.nl/esd/huidigebevragingen/v1

Reply:
waar wordt deze functionaliteit gevalideerd? Op productie of in acceptatie? Ik denk dat de urls daarop moet worden aangepast